src/hps_align/survey/_ballframe.py

```diff
@@ -0,0 +1,279 @@
+
+from ._utils import *
+from ._parser import Parser
+from ._cli import app
+
+
+class BallFrame:
+    """SVT uchannel ball frame class
+
+    Gets ball positions from survey data file and calculates basis vectors and origin.
+    The ball positions in the survey data file should be given in the OGP (or other global) system.
+
```

Comment:
You could also insert the image here? Again, not sure if it is helpful to have the image shown in multiple places in the website or if it just clutters the site. Would need to actually build the site locally to see what it looks like.

```suggestion
    Top
    ^^^
    .. image:: _static/uchannel_top.png
       :width: 600
       :alt: Diagram Showing Ball and Pin Frame Coordinate Systems for Top of UChannel

    Bottom
    ^^^^^^
    .. image:: _static/uchannel_bottom.png
       :width: 600
       :alt: Diagram Showing Ball and Pin Frame Coordinate Systems for Bottom of UChannel
```